components/calendar/src/chinese.rs

```diff
@@ -0,0 +1,994 @@
+        Self::months_for_every_year(date.0.year)
+    }
+}
+
+impl Date<Chinese> {
+    /// Construct a new Chinese date from a `year`, `month`, `leap_month`, and `day`.
+    /// `year` represents the Chinese year counted infinitely with -2636 (2637 BCE) as year Chinese year 1;
+    /// `month` represents the month of the year ordinally (ex. if it is a leap year, the last month will be 13, not 12);
+    /// `day` indicates the day of month
+    ///
+    /// ```rust
+    /// use icu::calendar::Date;
+    ///
+    /// let date_chinese = Date::try_new_chinese_date(4660, 6, 11)
+    ///     .expect("Failed to initialize Chinese Date instance.");
+    ///
+    /// assert_eq!(date_chinese.year().number, 4660);
+    /// assert_eq!(date_chinese.year().cyclic, Some(40));
+    /// assert_eq!(date_chinese.year().related_iso, Some(2023));
+    /// assert_eq!(date_chinese.month().ordinal, 6);
+    /// assert_eq!(date_chinese.day_of_month().0, 11);
+    /// ```
+    pub fn try_new_chinese_date(
+        year: i32,
+        month: u8,
+        day: u8,
+    ) -> Result<Date<Chinese>, CalendarError> {
+        ArithmeticDate::new_from_lunar_ordinals(year, month, day)
+            .map(ChineseDateInner)
+            .map(|inner| Date::from_raw(inner, Chinese))
+    }
+}
+
+impl DateTime<Chinese> {
+    /// Construct a new Chinese datetime from integers using the
+    /// -2636-based year system
+    ///
+    /// ```rust
+    /// use icu::calendar::DateTime;
+    ///
+    /// let chinese_datetime = DateTime::try_new_chinese_datetime(4660, 6, 11, 13, 1, 0)
+    ///     .expect("Failed to initialize Chinese DateTime instance.");
+    ///
+    /// assert_eq!(chinese_datetime.date.year().number, 4660);
+    /// assert_eq!(chinese_datetime.date.year().related_iso, Some(2023));
+    /// assert_eq!(chinese_datetime.date.year().cyclic, Some(40));
+    /// assert_eq!(chinese_datetime.date.month().ordinal, 6);
+    /// assert_eq!(chinese_datetime.date.day_of_month().0, 11);
+    /// assert_eq!(chinese_datetime.time.hour.number(), 13);
+    /// assert_eq!(chinese_datetime.time.minute.number(), 1);
+    /// assert_eq!(chinese_datetime.time.second.number(), 0);
+    /// ```
+    pub fn try_new_chinese_datetime(
+        year: i32,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
+    ) -> Result<DateTime<Chinese>, CalendarError> {
+        Ok(DateTime {
+            date: Date::try_new_chinese_date(year, month, day)?,
+            time: types::Time::try_new(hour, minute, second, 0)?,
+        })
+    }
+}
+
+impl ChineseBased for Chinese {
+    fn location(fixed: RataDie) -> Location {
+        let year = Iso::iso_from_fixed(fixed).year().number;
+        if year < 1929 {
+            CHINESE_LOCATION_PRE_1929
+        } else {
+            CHINESE_LOCATION_POST_1929
+        }
+    }
+
+    const EPOCH: RataDie = CHINESE_EPOCH;
+
+    fn new_chinese_based_date<Chinese>(year: i32, month: u8, day: u8) -> Date<Chinese> {
```

Comment:
This is defining a generic called `Chinese`, which can be anything, not just the struct `Chinese`.
```suggestion
    fn new_chinese_based_date(year: i32, month: u8, day: u8) -> Date<Chinese> {
```

Reply:
Ok thanks, I did this and moved the generic to part of the trait signature instead.